lib/Service/CallService.php

```diff
@@ -327,12 +330,23 @@ public function call(
 		$callLog->setStatusCode($data['response']['statusCode']);
 		$callLog->setStatusMessage($data['response']['statusMessage']);
 		$callLog->setRequest($data['request']);
-		$callLog->setResponse($data['response']);
 		$callLog->setCreated(new \DateTime());
 		$callLog->setExpires(new \DateTime('now + '.($data['response']['statusCode'] < 400 ? $source->getLogRetention() : $source->getErrorRetention()).' seconds'));
 
+		// Only persist response if we get bad requests or server errors.
+		if ($callLog->getStatusCode() >= 400 && $callLog->getStatusCode() < 600 || $logBody === true) {
+			$callLog->setResponse($data['response']);
+		} else {
+            $response = $data['response'];
+            unset($response['body']);
```

Comment:
Why do we have to unset here?

Reply:
In the first version, you did not write any data of the response (so even the response time wasn't logged). Instead, I now write the response without the body.